src/Security/MemberAuthenticator/Authenticator.php

```diff
@@ -1,32 +1,51 @@
 <?php
 
-namespace SilverStripe\Security;
+namespace SilverStripe\Security\MemberAuthenticator;
 
 use SilverStripe\Control\Controller;
 use SilverStripe\Control\Session;
 use SilverStripe\Forms\Form;
 use SilverStripe\ORM\ValidationResult;
 use InvalidArgumentException;
+use SilverStripe\Security\Authenticator as BaseAuthenticator;
+use SilverStripe\Security\Security;
+use SilverStripe\Security\Member;
+use SilverStripe\Security\LoginAttempt;
 
 /**
  * Authenticator for the default "member" method
  *
  * @author Markus Lanthaler <markus@silverstripe.com>
  */
-class MemberAuthenticator extends Authenticator
+class Authenticator implements BaseAuthenticator
 {
 
+    public function supportedServices()
+    {
+        // Bitwise-OR of all the supported services, to make a bitmask
+        return BaseAuthenticator::LOGIN | BaseAuthenticator::LOGOUT | BaseAuthenticator::CHANGE_PASSWORD
+            | BaseAuthenticator::RESET_PASSWORD | BaseAuthenticator::CMS_LOGIN;
+    }
+
     /**
-     * Contains encryption algorithm identifiers.
-     * If set, will migrate to new precision-safe password hashing
-     * upon login. See http://open.silverstripe.org/ticket/3004
-     *
-     * @var array
+     * @inherit
      */
-    private static $migrate_legacy_hashes = array(
-        'md5' => 'md5_v2.4',
-        'sha1' => 'sha1_v2.4'
-    );
+    public function authenticate($data, &$message)
+    {
+        $success = null;
+
+        // Find authenticated member
+        $member = $this->authenticateMember($data, $message, $success);
+
+        // Optionally record every login attempt as a {@link LoginAttempt} object
+        $this->recordLoginAttempt($data, $member, $success);
```

Comment:
I recall a comment around this also saying it would probably be better off in an Extension.

Is it an option to strip all these features out, into a separate "security-additions" kind of module that does these kind of things?